app/vmalert/static/js/custom.js

```diff
@@ -15,6 +17,84 @@ function toggleByID(id) {
     }
 }
 
+function debounce(func, delay) {
+    let timer;
+    return function(...args) {
+        clearTimeout(timer);
+        timer = setTimeout(() => {
+            func.apply(this, args);
+        }, delay);
+    };
+}
+
+$('#filter').on("keyup", debounce(filter, 500));
+
+function filter(){
+    $(".rule-table").removeClass('show');
+    $(".rule").show();
+    
+    if($("#filter").val().length === 0){
+        $(".group-heading").show()
+        return
+    }
+
+    $(".group-heading").hide()
+
+    filterRuleByName();
+    filterRuleByLabels();
+    filterGroupsByName();
+}
+
+function filterGroupsByName(){
+    $( ".group-heading" ).each(function() {
+        var groupName = $(this).attr('data-group-name');
+        var filter = $("#filter").val()
+
```

Comment:
One more little thing. you are using `let` in the parts of code, so I think it is better to use const instead of `var`.
Also please check your `let` usage. In some cases, as I can see you do not change the value, so probably it is `const` variable

Reply:
Nice tip!